src/index.js

```diff
@@ -983,7 +994,7 @@ export default class Carousel extends React.Component {
     const mouseEvents = this.getMouseEvents();
     const TransitionControl = Transitions[this.props.transitionMode];
     const validChildren = getValidChildren(this.props.children);
-
+    const orderedChildren = this.reverseChildren(children, rtl);
```

Comment:
this should be called with `validChildren` so that you aren't returning any invalid React children from the reverse method